### PR TITLE
Add Key / Colour Legend

### PR DESCRIPTION
**Pull request by lalitpatel**
### Summary

Add a Key / Legend on the left side bar.

#### Changes

- Creates a key in the left side bar showing various caret colors used in the tech tree.
- I have used two columns instead one to save some vertical space. 
- Use the `data.json` to fetch strings for the Key.
- Changes to the civ selector and civ logo to save vertical space and make it look cleaner.
- Changes to various paddings to save some verticals space and make it uniform across edges.

#### Other 
- Do not specify font family in SVG. Instead, use CSS to specify font for all the texts in SVG. This makes it easier to swap out fonts in the future.

### Screenshots
|Before | After |
| -- | -- |
| aoe2techtree net_ |  localhost_8000_ |

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -667,6 +679,29 @@
...
+            if (index % 2 === 0) {
+                tr = document.createElement('tr');
+            }
+            let td_color = document.createElement('td');
+            td_color.style.backgroundColor = legend[index]['colour'];
+            td_color.style.border = '1px outset #8a5d21';
+            tr.appendChild(td_color);
+            let td_type = document.createElement('td');
+            td_type.innerText = data.strings[data.tech_tree_strings[legend[index]['name']]];
+            // td_type.style.fontWeight = 'bold';
```
Comment (HSZemi) [inline index.html:697]: This looks like a leftover :slightly_smiling_face:

**Comment by pardoman** [comment]
> but it will not hurt either I guess

I fear that the localization buttons that I'm adding will not have enough space in there.

**Comment by lalitpatel** [comment]
@pardoman We can work out some design to add those buttons as well. I can also help push the #33 ahead. Are you on discord? I am `numb_aoe` on discord.

**Comment by HSZemi** [comment]
Let's look at that in #33. We can remove it again if we need the space for more important things.

**Comment by pardoman** [comment]
Sounds good.

**Pull request merged**